### PR TITLE
Change references to pkg.jenkins-ci.org to pkg.jenkins.io.

### PR DESCRIPTION
**Pull request by abayer**
cc @reviewbybees, @rtyler, @kohsuke, @daniel-beck

### CODE REVIEW

**Comment by svanoort** [comment]
Whoa keanu

**Comment by recena** [comment]
LGTM 👍

File: env/release.mk [hunk 1]
```diff
@@ -17,14 +17,14 @@ export RPMDIR=/srv/releases/jenkins/redhat${RELEASELINE}
 export SUSEDIR=/srv/releases/jenkins/opensuse${RELEASELINE}
 
 # where to put repository index and other web contents
-export  RPM_WEBDIR=/var/www/pkg.jenkins-ci.org.staging/redhat${RELEASELINE}
-export SUSE_WEBDIR=/var/www/pkg.jenkins-ci.org.staging/opensuse${RELEASELINE}
-export  DEB_WEBDIR=/var/www/pkg.jenkins-ci.org.staging/debian${RELEASELINE}
+export  RPM_WEBDIR=/var/www/pkg.jenkins.io.staging/redhat${RELEASELINE}
```
Comment (rtyler) [inline env/release.mk]: This part of the changes will break things unless accompanied by a change to [this script|https://github.com/jenkins-infra/jenkins-infra/blob/staging/dist/profile/files/mirrorbrain/sync.sh#L64]

Comment (abayer) [inline env/release.mk]: https://github.com/jenkins-infra/jenkins-infra/pull/508

**Comment by svanoort** [comment]
@abayer Is this intended to only apply to pre-jenkins 2 packaging?

**Comment by abayer** [comment]
Both. Lemme guess, I forgot that 2.x is on another branch. =)

**Comment by svanoort** [comment]
@abayer Nah, I assumed we were just going to merge/cherry-pick stuff if both needed it, was curious though.

**Comment by abayer** [comment]
Tough luck, I did a PR there too!

**Comment by svanoort** [comment]
@abayer How much do we want to redirect to jenkins.io vs. jenkins-ci.org?  There are a bunch of other references:  https://github.com/jenkinsci/packaging/search?utf8=%E2%9C%93&q=jenkins-ci.org

Not sure what the impact would be of changing all them though (case-by case basis there, some might be safest to leave in place).

**Comment by rtyler** [comment]
> Tough luck, I did a PR there too!

so brave

**Comment by svanoort** [comment]
@rtyler That gif makes my eyes bleed! 😆

**Comment by abayer** [comment]
@svanoort Piecemeal! There are probably some other places (i.e., bare references to `jenkins-ci.org`) that could/should be switched, but not everything actually has a `jenkins.io` equivalent yet. Therefore I just said "screw it, here's what's relevant for pkg, bugger everything else for now".

**Comment by svanoort** [comment]
@abayer Well, I guess that makes sense -- those other usages do need to be looked at.  🐛  for not changing the .htaccess files, in the subdirectories here, for example: https://github.com/jenkinsci/packaging/tree/master/env/release

**Comment by abayer** [comment]
Those refer to `mirrors.jenkins-ci.org`, not `pkg.jenkins-ci.org`. Separate change! Really, @rtyler should just go through here and change everything that's relevant. =)

**Comment by svanoort** [comment]
@abayer http://mirrors.jenkins.io/ exists -- I think if we change one, we should be changing both in the same PR.  Doing it this way is a bug, because then we're tying two of our separate domains together (and make the system more brittle -- any issues in the original site will potentially hit jenkins.io). 

Whatever the opshead's version of a bug is, it's that.

**Comment by rtyler** [comment]
FWIW, `pkg.jenkins-ci.org` is a CNAME of `pkg.jenkins.io` as is `mirrors.jenkins-ci.org` with regards to `mirrors.jenkins.io`

the `.io` domains became the "canonical" names when I migrated this infra to AWS a few weeks ago

**Comment by svanoort** [comment]
That's somewhat better that alternatives I could see, but I still think they should be changed in step (it doesn't make sense to do otherwise).

**Comment by abayer** [comment]
Happy now, @svanoort? Huh? Huh? =)

**Comment by svanoort** [comment]
@abayer  I :bee: lieve I can get behind this! :+1:  (yes. :smile: )

**Comment by rtyler** [comment]
So this should be merged yeah?

**Comment by svanoort** [comment]
@rtyler Double checking that we don't have someone wanting to freeze packaging for a release, then yes.

**Comment by rtyler** [comment]
@svanoort this code won't get execercised until @kohsuke creates a weekly release, or @olivergondza creates an RC right?

**Comment by svanoort** [comment]
@rtyler Correct.  Also I just merged this into the Jenkins 2 branch as well, so we're covered there.

**Pull request merged**